compiller/src/main/java/com/yheriatovych/reductor/processor/autoreducer/AutoReducerProcessingStep.java

```diff
@@ -62,20 +67,19 @@ private void emitGeneratedClass(StringReducerElement reducerElement, String pack
                 .superclass(TypeName.get(originalTypeElement.asType()));
 
         TypeName stateTypeName = TypeName.get(reducerElement.stateType);
+        MethodTypeInfo mdInfo = StateProperty.getReducerInterfaceReturnTypeInfo();
+        final TypeName reducerReturnTypeName = ParameterizedTypeName.get(
+                ClassName.get((Class<?>) ((ParameterizedType)mdInfo.getGenericReturnType()).getRawType()),
+                stateTypeName,
+                TypeName.get(mdInfo.getGenericsInReturnType()[1])
+        );
         MethodSpec.Builder reduceMethodBuilder = MethodSpec.methodBuilder("reduce")
                 .addModifiers(Modifier.PUBLIC)
-                .returns(stateTypeName)
+                .returns(reducerReturnTypeName)
                 .addAnnotation(Override.class)
                 .addParameter(stateTypeName, "state")
                 .addParameter(Action.class, "action");
 
-        if (reducerElement.initMethod != null) {
-            reduceMethodBuilder.beginControlFlow("if (state == null)")
-                    .addStatement("state = $N()", reducerElement.initMethod.getName())
-                    .endControlFlow()
-                    .addCode("\n");
-        }
```

Comment:
Not handle `initMethod` anymore. Because it's hard or impossible to generate getter and setting code in annotation of `AutoReducer`. Annotation only can handle the component with annotation mark. Annotation cannot get info that beyond the component.